# Module 1 Lab 3: Choosing an Insurance Provider

In [1]:
import pandas as pd
data = { "customer":
[1,2,3,4,5,6,7,8,9], "provider":
["GEICO","GEICO","GEICO","Progressive","Progressive","Progressive","USAA","USAA","USAA"], 
         "rating": [4.7,8.3,9.2,7.4,6.7,8.9,3.8,6.3,8.1], } 
df = pd.DataFrame(data) 
df                              

,customer,provider,rating
0,1,GEICO,4.7
1,2,GEICO,8.3
2,3,GEICO,9.2
3,4,Progressive,7.4
4,5,Progressive,6.7
5,6,Progressive,8.9
6,7,USAA,3.8
7,8,USAA,6.3
8,9,USAA,8.1


In [2]:
summary = df.groupby("provider")["rating"].agg(["min","mean","max"]).round(2); summary

,min,mean,max
provider,,,
GEICO,4.7,7.40,9.2
Progressive,6.7,7.67,8.9
USAA,3.8,6.07,8.1


USAA's minimum rating is 3.8 and GEICO is only 0.9 better at 4.7. These are indicative of significant negative customer experience. Progressive has the highest minimum rating by far at 6.7, higher than USAA's mean of 6.07. This indicates that the worst customer experiences with Progressive are better than mean customer experiences with USAA. While GEICO has the highest max of 9.2, it is close with Progressive which is at 8.9. Progressive is clearly the lowest risk given it holds the highest minimum and mean, while still being high reward given its maximum. I would choose Progressive.